heron/cli/src/python/argparser.py

```diff
@@ -0,0 +1,143 @@
+	@classmethod
+	def initializeFromRC(hrc):
+		if  len(hrc.cmdmap) > 0:
+			return 
+		if os.path.exists(HERON_RC):
+
+			with open(HERON_RC) as f:
+				hrc.cmdmap['*']['*'] = ''
+				for line in f:
+					m = p.match(line)
+					value = ''
+					command=''
+					env = ''
+					if (m != None):
+						value = hrc.remove_comments(m.group(3).rstrip(os.linesep))
+						command = ( m.group(1),'') [ m.group(1) == None or m.group(1) == '' ]
+						env = ( m.group(2), '') [ m.group(2) == None or m.group(2) == '']
+					else:
+						value = hrc.remove_comments(line.rstrip(os.linesep))
+
+					args_list = args.insert_bool_values(value.split())  # make sure that all the single args have a boolean value associated so that we can load the args to a key value structure
+					args_list_string = ' '.join (args_list)
+					if command == None or command == '' :
+						continue
+					if ( command in hrc.cmdmap and env in hrc.cmdmap[command]):
+						hrc.cmdmap[command][env] =  hrc.cmdmap[command][env] 	+ ' ' + args_list_string	
+					else:
+						hrc.cmdmap[command][env] =  args_list_string
+			#print "cmdmap", hrc.cmdmap	
+		else:
+			print "WARN: %s is not an existing file" % HERON_RC
+
+
+    # for each command / cluster-role-env combination, get the commands from heronrc
+    #     remove any duplicates that have already been supplied already  and present in the command-dictionary     
+	@classmethod
+	def get_args_for_command_role (hrc, command, role):
+		args_for_command_role=''
+
+		if ( command in hrc.cmdmap and role in hrc.cmdmap[command]):
+			args_for_command_role = (hrc.cmdmap[command][role],args_for_command_role) [hrc.cmdmap[command][role] == None or hrc.cmdmap[command][role] == '']
+		return args_for_command_role.split()
+
+	# this is invoked when the parser.parse_args is called
+	#  apply the commands in the following precedence order
+	#  use the defaults in the command line
+
+	def _read_args_from_files(self,arg_strings):
+		new_arg_strings = []
+		command = sys.argv[1]
+		role = sys.argv[2]
+		new_arg_strings.extend(self.get_args_for_command_role('*', '*'))
+		new_arg_strings.extend(self.get_args_for_command_role(command, '*'))
+		new_arg_strings.extend(self.get_args_for_command_role(command, role))
+		new_arg_strings.extend(arg_strings)
+		return arg_strings
+
+#test stub
+def main():
+	#parser = HeronRCArgumentParser(description="test parser",fromfile_prefix_chars='@')
+	  #parser = argparse.ArgumentParser(
+	parser = HeronRCArgumentParser(
+		prog = 'heron',
+		epilog = help_epilog,
+		formatter_class=SubcommandHelpFormatter,
+		fromfile_prefix_chars='@',
+		add_help = False)
+	subparsers = parser.add_subparsers(
+		title = "Available commands", 
+		metavar = '<command> <options>')
+
+	activate.create_parser(subparsers)
+	args, unknown_args = parser.parse_known_args()
+	print args, unknown_args
+
+
+if __name__ == "__main__":
+	sys.exit(main())
```

Comment:
why `argparser` module can be invoked as a Python program? If this is for test purpose, please do not run test manually and add test under `heron/cli/tests` to test this module.